group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/GroupSpec.java

```diff
@@ -26,9 +26,9 @@
  */
 public interface GroupSpec {
     /**
-     * @return Member metadata keyed by member Id.
+     * @return Member subscription metadata keyed by member Id.
      */
-    Map<String, MemberSubscriptionSpec> members();
+    Map<String, MemberSubscriptionSpec> memberSubscriptions();
```

Comment:
I think that we also wanted to return a `Collection` here.

Reply:
Okay. If we use it then we must keep it.

Reply:
An alternative would be to provide methods for the access patterns that we need (e.g. memberIds(), subscription(memberId), etc.). I need to think about it.

Reply:
Yeah I thought about it too, we could have memberIds() in the groupSpec, but even that would have to iterate through the collection and collect them. Lmk if you have some good ideas

Reply:
Hum... I don't think so. If we do this, we would still use a Map to back those methods in the implementation.

Reply:
I was talking about the case when we would keep it as a Collection

Reply:
discussed offline